airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQueryDestinationHandler.java

```diff
@@ -47,13 +49,45 @@ public void execute(final String sql) throws InterruptedException {
 
     Job job = bq.create(JobInfo.of(QueryJobConfiguration.newBuilder(sql).build()));
     job = job.waitFor();
+    // waitFor() seems to throw an exception, but javadoc says we're supposed to handle this case
+    if (job.getStatus().getError() != null) {
+      throw new RuntimeException(job.getStatus().getError().toString());
+    }
 
     JobStatistics.QueryStatistics statistics = job.getStatistics();
-    LOGGER.info("Completed sql {} in {} ms; processed {} bytes; billed for {} bytes",
+    LOGGER.info("Root-level job {} completed in {} ms; processed {} bytes; billed for {} bytes",
         queryId,
         statistics.getEndTime() - statistics.getStartTime(),
         statistics.getTotalBytesProcessed(),
         statistics.getTotalBytesBilled());
+
+    // SQL transactions can spawn child jobs, which are billed individually. Log their stats too.
+    if (statistics.getNumChildJobs() != null) {
+      // There isn't (afaict) anything resembling job.getChildJobs(), so we have to ask bq for them
+      bq.listJobs(BigQuery.JobListOption.parentJobId(job.getJobId().getJob())).streamAll()
+          .sorted(Comparator.comparing(childJob -> childJob.getStatistics().getEndTime()))
+          .forEach(childJob -> {
+            JobConfiguration configuration = childJob.getConfiguration();
+            if (configuration instanceof QueryJobConfiguration qc) {
+              JobStatistics.QueryStatistics childQueryStats = childJob.getStatistics();
+              String truncatedQuery = qc.getQuery()
```

Comment:
nit: maybe add "..." to the end of the string if we truncate for clarity?

Reply:
done https://github.com/airbytehq/airbyte/pull/28373/commits/b283cae6bc897a90c663af9853a1c073c81c484d